dockstore-common/src/main/java/io/dockstore/common/NextflowUtilities.java

```diff
@@ -0,0 +1,150 @@
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package io.dockstore.common;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.StringReader;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Properties;
+
+import com.google.common.base.Joiner;
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.ConfigurationConverter;
+import org.apache.commons.configuration2.INIConfiguration;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class NextflowUtilities {
+    private static final Logger LOG = LoggerFactory.getLogger(NextflowUtilities.class);
+    private static final String DEFAULT_NEXTFLOW_VERSION = "19.01.0";
+
+    private NextflowUtilities() {
+        // hide the default constructor for a utility class
+    }
+
+    public static File getNextFlowTargetFile(INIConfiguration config) {
+        String nextflowVersion = config.getString("nextflow-version", DEFAULT_NEXTFLOW_VERSION);
+        return getNextFlowTargetFile(nextflowVersion);
+    }
+
+    private static File getNextFlowTargetFile() {
+        return getNextFlowTargetFile(DEFAULT_NEXTFLOW_VERSION);
+    }
+
+    private static File getNextFlowTargetFile(String nextflowVersion) {
+        String nextflowExec =
+            "https://github.com/nextflow-io/nextflow/releases/download/v" + nextflowVersion + "/nextflow-" + nextflowVersion + "-all";
+        if (!Objects.equals(DEFAULT_NEXTFLOW_VERSION, nextflowVersion)) {
+            System.out.println("Running with Nextflow " + nextflowVersion + " , Dockstore tests with " + DEFAULT_NEXTFLOW_VERSION);
+        }
+
+        // grab the nextflow jar if needed
+        String libraryLocation =
+            System.getProperty("user.home") + File.separator + ".dockstore" + File.separator + "libraries" + File.separator;
+        URL nextflowURL;
+        String nextflowFilename;
+        try {
+            nextflowURL = new URL(nextflowExec);
+            nextflowFilename = new File(nextflowURL.toURI().getPath()).getName();
+        } catch (MalformedURLException | URISyntaxException e) {
+            throw new NextflowParsingException("Could not create Nextflow location", e);
+        }
+        String nextflowTarget = libraryLocation + nextflowFilename;
+        File nextflowTargetFile = new File(nextflowTarget);
+        if (!nextflowTargetFile.exists()) {
+            try {
+                FileUtils.copyURLToFile(nextflowURL, nextflowTargetFile);
```

Comment:
Looks like a decent sized file -- would it be worth it to print a downloading message to the user?

Reply:
Added to #1520 (probably same painpoint)